distributed/semaphore.py

```diff
@@ -99,6 +99,10 @@ def create(self, comm=None, name=None, max_leases=None):
         if name not in self.max_leases:
             assert isinstance(max_leases, int), max_leases
             self.max_leases[name] = max_leases
+            # Initialize metrics
```

Comment:
I just added this to make sure all metrics are set when the semaphore is created. @fjetter I could also move this to a new PR to not block the commits that you have already reviewed

Reply:
This logic somehow defeats the purpose of using defaultdicts, doesn't it? If we initialize these data structure here explicitly, we don't need defaultdicts.
This feels rather like a shortcoming of the prometheus collector and I don't understand why this is needed. If the collector calls `len(self.leases[name])` or something similar this should be properly handled already, shouldn't it?

Reply:
Sure, I'll move this logic to the prometheus collector. I'll remove the commit for now